src/lib/reactor/fiber_lib_boost.cpp

```diff
@@ -66,12 +67,13 @@ void FiberManagerLib::shared_mutex::lock() {
         } else {
             // Owned by the writer or reader, add ourselves to writer queue
             if (s_writer_priority) {
-                ++m_write_waiters;
+                if (!already_waiter) { ++m_write_waiters; }

```

Comment:
Why do you need another already_waiter variable? Can't this be simply
```suggestion
                if (!am_i_write_waiter) { 
                     ++m_write_waiters;
                     am_i_write_waiter = true;
                 }
```